src/App.tsx

```diff
@@ -15,49 +16,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  Reset = 'reset',
+}
+
 export const App: React.FC = () => {
+  const [viewGoods, setViewGoods] = useState(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType | null>(null); // Permite null para indicar nenhum sort aplicado
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortByAlphabetic = () => {
+    setSortType(SortType.Alphabetically);
+    const sortedGoods = [...goodsFromServer].sort((good1, good2) =>
+      good1.localeCompare(good2),
+    );
+
+    setViewGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.Length);
+    const sortedGoods = [...goodsFromServer].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+
+    setViewGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+  };
+
+  const sortReverse = () => {
+    setIsReversed(prev => !prev);
+    setViewGoods([...viewGoods].reverse());
+  };
+
+  const sortReset = () => {
+    setSortType(SortType.Reset);
+    setIsReversed(false);
+    setViewGoods(goodsFromServer);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info${
+            sortType !== SortType.Alphabetically ? ' is-light' : ''
+          }`}
+          onClick={sortByAlphabetic}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success${sortType !== SortType.Length ? ' is-light' : ''}`}
+          onClick={sortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning${!isReversed ? ' is-light' : ''}`}
+          onClick={sortReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {viewGoods.join() !== goodsFromServer.join() && (
```

Comment:
The condition `viewGoods.join() !== goodsFromServer.join()` might not work as expected if the order of `viewGoods` is the same as `goodsFromServer` but reversed. Consider using a more robust method to check if the list has been altered, such as comparing the arrays directly or using a flag to track changes.